### PR TITLE
unifying vertica connector as jdbc connector

### PR DESCRIPTION
**Pull request by Trianz-Akshay**
*Description of changes:*
As suggested in qpt review comment, unified the vertical connector code as jdbc connector. 
There are some changes in connector documentation as part of this PR. I will share doc separately. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java [hunk 1]
```diff
@@ -367,11 +328,14 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request
      */
     @Override
     public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest request)
-            throws SQLException
     {
```
Comment (AbdulR3hman) [inline athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java:331]: it doesn't seem we are handling splits from QPT? Is that on purpose?

Comment (AbdulR3hman) [reply athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java:331]: you could use: `setupQueryPassthroughSplit` from the parent class.

**Review by AbdulR3hman — CHANGES_REQUESTED** [review]

File: athena-vertica/src/test/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandlerTest.java [hunk 2]
```diff
@@ -159,39 +178,20 @@ public void doListSchemaNames() throws SQLException
...
+                    new ListSchemasRequest(this.federatedIdentity,
+                            "testQueryId", "testCatalog"));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
         Assert.assertArrayEquals(expected, listSchemasResponse.getSchemas().toArray());
     }
 
-    @Test
-    public void doListTables() throws SQLException {
```
[the remark this answers is not in the document]
Comment (Trianz-Akshay) [reply athena-vertica/src/test/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandlerTest.java:172]: added.

File: athena-vertica/src/test/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandlerTest.java [hunk 3]
```diff
@@ -293,24 +293,23 @@ public void getPartitions() throws Exception
             assertTrue(partitions.getRowCount() > 0);
             logger.info("doGetTableLayout: partitions[{}]", partitions.getRowCount());
 
+        } finally {
```
[the remark this answers is not in the document]
Comment (Trianz-Akshay) [reply athena-vertica/src/test/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandlerTest.java]: refactored for try with resource.

File: athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java (hunk 1 above)
Comment (Trianz-Akshay) [reply athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaMetadataHandler.java:330]: here there is a somewhat different approach than other JDBC connectors. creating SQL statement from qpt args and exporting results to S3.

**Review by AbdulR3hman — APPROVED** [review]

**Pull request merged**